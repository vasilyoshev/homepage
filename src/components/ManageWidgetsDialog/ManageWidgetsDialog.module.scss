.addWidgetsDialogTitle {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.dialogContent {
  display: flex;
  flex-wrap: wrap;
}

.widgetPreviewCard {
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 100px;
  height: 100px;
  margin: 5px;
}

.widgetIcon {
  width: 50px !important;
  height: 50px !important;
}

.addWidgetsDialogPaper {
  width: 600px;
}

.addWidgetsActionAreaRoot {
  display: flex !important;
  height: 100%;
}
