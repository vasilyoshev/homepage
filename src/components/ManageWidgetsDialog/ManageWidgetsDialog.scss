.add-widgets-dialog-title {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.dialog-content {
  display: flex;
  flex-wrap: wrap;
}

.widget-preview-card {
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 100px;
  height: 100px;
  margin: 5px;
}

.widget-icon {
  width: 50px !important;
  height: 50px !important;
}

.add-widgets-dialog-paper {
  width: 600px;
}

.add-widgets-action-area-root {
  display: flex !important;
  height: 100%;
}
