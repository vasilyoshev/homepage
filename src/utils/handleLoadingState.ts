import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
