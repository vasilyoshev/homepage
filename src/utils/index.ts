export * from './handleLoadingState';
export * from './testUtils';
