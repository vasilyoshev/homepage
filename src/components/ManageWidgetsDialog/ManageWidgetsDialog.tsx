import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Card, CardActionArea, Dialog, DialogContent, DialogTitle, IconButton, Typography } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import CalendarTodayIcon from '@material-ui/icons/CalendarToday';
import { selectIsCalendarWidgetActive, toggleWidget } from 'slices';
import './ManageWidgetsDialog.scss';

type ManageWidgetsDialogType = {
  isOpen: boolean;
  onClose: () => void;
};

export const ManageWidgetsDialog: React.FC<ManageWidgetsDialogType> = ({ isOpen, onClose }) => {
  const dispatch = useDispatch();
  const isCalendarWidgetActive = useSelector(selectIsCalendarWidgetActive);

  const toggleWidgetByName = (widgetName: string): void => {
    dispatch(toggleWidget(widgetName));
  };

  return (
    <Dialog onClose={onClose} open={isOpen} classes={{ paper: 'add-widgets-dialog-paper' }}>
      <DialogTitle disableTypography className="add-widgets-dialog-title">
        <Typography variant="h5">Manage Widgets</Typography>
        <IconButton onClick={onClose}>
          <CloseIcon />
        </IconButton>
      </DialogTitle>
      <DialogContent className="dialog-content">
        <Card className="widget-preview-card" onClick={() => toggleWidgetByName('calendar')}>
          <CardActionArea classes={{ root: 'add-widgets-action-area-root' }}>
            <CalendarTodayIcon color={isCalendarWidgetActive ? 'primary' : 'secondary'} className="widget-icon" />
          </CardActionArea>
        </Card>
      </DialogContent>
    </Dialog>
  );
};
