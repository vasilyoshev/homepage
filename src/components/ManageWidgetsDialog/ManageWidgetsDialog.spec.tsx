import { render } from 'utils';
import { ManageWidgetsDialog } from './ManageWidgetsDialog';

describe('ManageWidgetsDialog', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<ManageWidgetsDialog isOpen={true} onClose={() => null} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
