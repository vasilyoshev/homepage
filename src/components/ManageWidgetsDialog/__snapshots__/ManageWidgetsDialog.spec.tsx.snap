// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageWidgetsDialog renders successfully 1`] = `<DocumentFragment />`;
